protocol/oracle-manager/contracts/utils/ChainlinkNodeLibrary.sol

```diff
@@ -6,27 +6,31 @@ import "../interfaces/external/IAggregatorV3Interface.sol";
 
 library ChainlinkNodeLibrary {
     function process(bytes memory parameters) internal view returns (Node.Data memory) {
-        (address chainlinkAggr, uint twapTimeInterval) = abi.decode(parameters, (address, uint));
+        (address chainlinkAggr, uint256 twapTimeInterval, uint8 decimals) = abi.decode(
+            parameters,
+            (address, uint256, uint8)
+        );
 
         (uint80 roundId, int256 price, , uint256 updatedAt, ) = IAggregatorV3Interface(
             chainlinkAggr
         ).latestRoundData();
         int256 finalPrice = twapTimeInterval == 0
             ? price
             : getTwapPrice(chainlinkAggr, roundId, price, twapTimeInterval);
-        return Node.Data(finalPrice, updatedAt, 0, 0);
+
+        return Node.Data(upscale(finalPrice, 18 - decimals), updatedAt, 0, 0);
```

Comment:
nice

Reply:
@Rickk137 does this line work if the decimals is greater than 18? `upscale` takes a uint.